doc/console/SONiC-Console-Switch-High-Level-Design.md

```diff
@@ -0,0 +1,361 @@
+
+# SONiC Console Switch
+
+# High Level Design Document
+
+#### Revision 0.1
+
+# Table of Contents
+
+# List of Tables
+
+# Revision
+| Rev |     Date    |          Authors             | Change Description                |
+|:---:|:-----------:|:----------------------------:|-----------------------------------|
+| 0.1 | 08/20/2020  |  Jing Kan | Initial version                   |
+
+# About this Manual
+
+This document provides general information about Console switch features implementaion in SONiC
+
+# Scope
+
+This document describes the functionality and high level design of the Console switch features in SONiC.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| Term | Meaning
+|---|---|
+| SSH | Secure Shell |
+
+# 1 Feature Overview
+Console Switch enable user to manage routers/switchs via console port.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+1. Support configuration of flow control on each console port
+1. Support configuration of baudrate on each console port
+1. Support access console line via reverse SSH
+
+### 1.1.2 Configuration and Management Requirements
+
+1. Support configuration using SONiC CONFIG_DB
+1. Support configuration using SONiC MINIGRAPH
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+
+Console device(Serial Hub) connect to a SONiC device via USB link.
```

Comment:
That's a good idea! Let me try make an abstraction design and document it in this HLD

Reply:
I am thinking the abstraction, for current command line tool, the flow is:
```
user -> consutil <command> -> picocom -> ttyUSB
```
The question is how the vendor provider them own implementation?
We can simply abstract this part```-> picocom -> ttyUSB``` with below three interface:
```
input: baud rate, flow control, line number

output for connect: interactive shell session
output for show: is busy or not
output for clear: success or not
``` 
But I am not sure where to put the implementation and how SONiC OS to decide which implementation should be use.
Are we able to get he HwSKU and use this information to decide which implementation will be use?

Reply:
We can put the hw_sku (console device) into the CONSOLE_PORT table for each port since we might connect multiple different console devices. Then we can decide which driver will be used when user execute consutil command.

The renaming question is that who has the responsibility to implement the abstraction? It is not a part of SAI, i guess we need to implement it if we want to support specified console device.